docs/contributing/github_integration.rst

```diff
@@ -0,0 +1,115 @@
+GitHub Integration
+====================
+
+buildtest has several github integration, including automated checks during PR that maintainers will check to
+during the PR review. You should check results from the `buildtest actions <https://github.com/HPC-buildtest/buildtest-framework/actions>`_
+to see if any fixes need to me made.
+
+It's good practice to check Travis `builds <https://travis-ci.com/HPC-buildtest/buildtest-framework>`_ since we use Travis
+to run regression test and Codecov and Coveralls depend on Travis to pass all checks.
+
+You will want to make sure code is formatted via black as we have automated checks for python formatting. If you have not
+setup the black hook check out :ref:`black_hook`
```

Comment:
Maybe for now we can add something like:

> If you run the black linter and find that the check still fails, check the step of the action that installs black and ensure that you are using the same version locally.

Another thing we can do is to pin a particular black version and then tell the user up front to use it, also spit out a warning in the commit hook if it's a different version.

Reply:
If we want to pin a version, it may make sense to add this to `docs/requirements.txt` so that all deps are added in one place and user dont have to do this manually. Looking at https://buildtest.readthedocs.io/en/devel/contributing.html#github-hooks i can see the `pip install black==19.3b0` be something user don't have to worry about if its part of `requirements.txt`. Anyhow I will add some section about your comment in the docs.

Reply:
Yeah this is a good idea. Then we will need to install that version explicitly for the test as well!